ng server does not work if the folder path has special character eg '!'

> Please provide us with the following information:
> ---------------------------------------------------------------

### OS?
> Mac OS Sierra 10.12.1


### Versions.
> angular-cli: 1.0.0-beta.19-3
> node: v6.9.1


### Repro steps.
1. Create a folder named "!doesnotwork"
2. Cd \!doesnotwork
3. ng new no-property
4. cd no-property
5. ng serve

It failing reporting missing files.


### The log given by the failure.
`bash-3.2$ ng new no-peoperty
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
installing ng2
  create .editorconfig
  create README.md
  create src/app/app.component.css
  create src/app/app.component.html
  create src/app/app.component.spec.ts
  create src/app/app.component.ts
  create src/app/app.module.ts
  create src/app/index.ts
  create src/assets/.gitkeep
  create src/environments/environment.prod.ts
  create src/environments/environment.ts
  create src/favicon.ico
  create src/index.html
  create src/main.ts
  create src/polyfills.ts
  create src/styles.css
  create src/test.ts
  create src/tsconfig.json
  create src/typings.d.ts
  create angular-cli.json
  create e2e/app.e2e-spec.ts
  create e2e/app.po.ts
  create e2e/tsconfig.json
  create .gitignore
  create karma.conf.js
  create package.json
  create protractor.conf.js
  create tslint.json
Installing packages for tooling via npm.
Installed packages for tooling via npm.
bash-3.2$ cd no-peoperty/
bash-3.2$ ng serve
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
** NG Live Development Server is running on http://localhost:4200. **
761ms building modules                  1ms add0m143ms c38ms asset5ms emitting gHash: 4c60eeca27045fcc7984
Version: webpack 2.1.0-beta.25
Time: 1019ms
           Asset       Size  Chunks             Chunk Names
  main.bundle.js     244 kB    0, 2  [emitted]  main
styles.bundle.js  303 bytes    1, 2  [emitted]  styles
       inline.js    5.53 kB       2  [emitted]  inline
        main.map     290 kB    0, 2  [emitted]  main
      styles.map   93 bytes    1, 2  [emitted]  styles
      inline.map    5.59 kB       2  [emitted]  inline
      index.html    5.49 kB          [emitted]  

ERROR in multi main
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/angular-cli/models'
 @ multi main

ERROR in multi styles
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/angular-cli/models'
 @ multi styles

ERROR in   Error: Child compilation failed:
  Entry module not found: Error: Can't resolve '/Users/Subhasish/Documents/eShar  k/Work/RnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peope  rty/node_modules/angular-cli/models':
  Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/  Subhasish/Documents/eShark/Work/RnD/!Property/no-peoperty/node_modules/angular  -cli/models'
  
  - compiler.js:76 
    [no-peoperty]/[html-webpack-plugin]/lib/compiler.js:76:16
  
  - Compiler.js:279 Compiler.<anonymous>
    [no-peoperty]/[webpack]/lib/Compiler.js:279:10
  
  - Compiler.js:474 
    [no-peoperty]/[webpack]/lib/Compiler.js:474:13
  
  - Tapable.js:81 next
    [no-peoperty]/[tapable]/lib/Tapable.js:81:11
  
  - CachePlugin.js:61 Compiler.<anonymous>
    [no-peoperty]/[webpack]/lib/CachePlugin.js:61:4
  
  - Tapable.js:85 Compiler.applyPluginsAsync
    [no-peoperty]/[tapable]/lib/Tapable.js:85:13
  
  - Compiler.js:471 Compiler.<anonymous>
    [no-peoperty]/[webpack]/lib/Compiler.js:471:10
  
  - Tapable.js:73 Compilation.applyPluginsAsync
    [no-peoperty]/[tapable]/lib/Tapable.js:73:70
  
  - Compilation.js:626 Compilation.<anonymous>
    [no-peoperty]/[webpack]/lib/Compilation.js:626:18
  
  - Tapable.js:73 Compilation.applyPluginsAsync
    [no-peoperty]/[tapable]/lib/Tapable.js:73:70
  
  - Compilation.js:617 Compilation.<anonymous>
    [no-peoperty]/[webpack]/lib/Compilation.js:617:10
  
  - Tapable.js:73 Compilation.applyPluginsAsync
    [no-peoperty]/[tapable]/lib/Tapable.js:73:70
  
  - Compilation.js:612 Compilation.<anonymous>
    [no-peoperty]/[webpack]/lib/Compilation.js:612:9
  
  - Tapable.js:73 Compilation.applyPluginsAsync
    [no-peoperty]/[tapable]/lib/Tapable.js:73:70
  
  - Compilation.js:608 Compilation.<anonymous>
    [no-peoperty]/[webpack]/lib/Compilation.js:608:8
  
  - Tapable.js:73 Compilation.applyPluginsAsync
    [no-peoperty]/[tapable]/lib/Tapable.js:73:70
  
  - Compilation.js:554 Compilation.seal
    [no-peoperty]/[webpack]/lib/Compilation.js:554:7
  
  - Compiler.js:468 Compiler.<anonymous>
    [no-peoperty]/[webpack]/lib/Compiler.js:468:16
  
  - Tapable.js:152 
    [no-peoperty]/[tapable]/lib/Tapable.js:152:11
  
  - Compilation.js:452 Compilation.<anonymous>
    [no-peoperty]/[webpack]/lib/Compilation.js:452:10
  
  - Compilation.js:347 Compilation.errorAndCallback
    [no-peoperty]/[webpack]/lib/Compilation.js:347:3
  
  - Compilation.js:364 Compilation.<anonymous>
    [no-peoperty]/[webpack]/lib/Compilation.js:364:11
  
  - NormalModuleFactory.js:40 onDoneResolving
    [no-peoperty]/[webpack]/lib/NormalModuleFactory.js:40:20
  
  - NormalModuleFactory.js:117 
    [no-peoperty]/[webpack]/lib/NormalModuleFactory.js:117:20
  
  - async.js:726 
    [no-peoperty]/[webpack]/[async]/lib/async.js:726:13
  
  - async.js:52 
    [no-peoperty]/[webpack]/[async]/lib/async.js:52:16
  
  - async.js:241 done
    [no-peoperty]/[webpack]/[async]/lib/async.js:241:17
  
  - async.js:44 
    [no-peoperty]/[webpack]/[async]/lib/async.js:44:16
  
  - async.js:723 
    [no-peoperty]/[webpack]/[async]/lib/async.js:723:17
  
  - async.js:167 
    [no-peoperty]/[webpack]/[async]/lib/async.js:167:37
  
  - async.js:52 
    [no-peoperty]/[webpack]/[async]/lib/async.js:52:16
  
  - async.js:361 
    [no-peoperty]/[webpack]/[async]/lib/async.js:361:13
  
  - async.js:52 
    [no-peoperty]/[webpack]/[async]/lib/async.js:52:16
  
  - async.js:241 done
    [no-peoperty]/[webpack]/[async]/lib/async.js:241:17
  
  - async.js:44 
    [no-peoperty]/[webpack]/[async]/lib/async.js:44:16
  
  - async.js:358 
    [no-peoperty]/[webpack]/[async]/lib/async.js:358:17
  
  - NormalModuleFactory.js:216 
    [no-peoperty]/[webpack]/lib/NormalModuleFactory.js:216:19
  
  - Resolver.js:70 onResolved
    [no-peoperty]/[enhanced-resolve]/lib/Resolver.js:70:11
  
  - createInnerCallback.js:31 loggingCallbackWrapper
    [no-peoperty]/[enhanced-resolve]/lib/createInnerCallback.js:31:19
  
  - Resolver.js:138 afterInnerCallback
    [no-peoperty]/[enhanced-resolve]/lib/Resolver.js:138:10
  
  - createInnerCallback.js:31 loggingCallbackWrapper
    [no-peoperty]/[enhanced-resolve]/lib/createInnerCallback.js:31:19
  
  - Tapable.js:108 Resolver.applyPluginsAsyncSeriesBailResult1
    [no-peoperty]/[tapable]/lib/Tapable.js:108:46
  
  - Resolver.js:125 innerCallback
    [no-peoperty]/[enhanced-resolve]/lib/Resolver.js:125:19
  
  - createInnerCallback.js:31 loggingCallbackWrapper
    [no-peoperty]/[enhanced-resolve]/lib/createInnerCallback.js:31:19
  
  - Tapable.js:210 
    [no-peoperty]/[tapable]/lib/Tapable.js:210:15
  
  - UnsafeCachePlugin.js:39 
    [no-peoperty]/[enhanced-resolve]/lib/UnsafeCachePlugin.js:39:4
  
  - createInnerCallback.js:31 loggingCallbackWrapper
    [no-peoperty]/[enhanced-resolve]/lib/createInnerCallback.js:31:19
  
  - Resolver.js:138 afterInnerCallback
    [no-peoperty]/[enhanced-resolve]/lib/Resolver.js:138:10
  
  - createInnerCallback.js:31 loggingCallbackWrapper
    [no-peoperty]/[enhanced-resolve]/lib/createInnerCallback.js:31:19
  
  - Tapable.js:108 Resolver.applyPluginsAsyncSeriesBailResult1
    [no-peoperty]/[tapable]/lib/Tapable.js:108:46
  
  - Resolver.js:125 innerCallback
    [no-peoperty]/[enhanced-resolve]/lib/Resolver.js:125:19
  


ERROR in ./~/sockjs-client/lib/entry.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib'
 @ ./~/sockjs-client/lib/entry.js 1:0-114
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/punycode/punycode.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/punycode'
 @ ./~/punycode/punycode.js 1:0-114
 @ ./~/url/url.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/main.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib'
 @ ./~/sockjs-client/lib/main.js 1:0-107
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/main.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib'
 @ ./~/sockjs-client/lib/main.js 1:0-114
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/location.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib'
 @ ./~/sockjs-client/lib/location.js 1:0-114
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/info-receiver.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib'
 @ ./~/sockjs-client/lib/info-receiver.js 1:0-107
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/iframe-bootstrap.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib'
 @ ./~/sockjs-client/lib/iframe-bootstrap.js 1:0-107
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/utils/url.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/utils'
 @ ./~/sockjs-client/lib/utils/url.js 1:0-107
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/utils/event.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/utils'
 @ ./~/sockjs-client/lib/utils/event.js 1:0-114
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/utils/browser.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/utils'
 @ ./~/sockjs-client/lib/utils/browser.js 1:0-114
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/utils/transport.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/utils'
 @ ./~/sockjs-client/lib/utils/transport.js 1:0-107
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/utils/log.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/utils'
 @ ./~/sockjs-client/lib/utils/log.js 1:0-114
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/websocket.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport'
 @ ./~/sockjs-client/lib/transport/websocket.js 1:0-107
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/xhr-streaming.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport'
 @ ./~/sockjs-client/lib/transport/xhr-streaming.js 1:0-114
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/jsonp-polling.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport'
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js 1:0-114
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/lib/iframe-wrap.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/lib'
 @ ./~/sockjs-client/lib/transport/lib/iframe-wrap.js 1:0-114
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/info-iframe.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib'
 @ ./~/sockjs-client/lib/info-iframe.js 1:0-107
 @ ./~/sockjs-client/lib/info-receiver.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/info-iframe.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib'
 @ ./~/sockjs-client/lib/info-iframe.js 1:0-114
 @ ./~/sockjs-client/lib/info-receiver.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/info-ajax.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib'
 @ ./~/sockjs-client/lib/info-ajax.js 1:0-107
 @ ./~/sockjs-client/lib/info-receiver.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/utils/iframe.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/utils'
 @ ./~/sockjs-client/lib/utils/iframe.js 1:0-107
 @ ./~/sockjs-client/lib/iframe-bootstrap.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/utils/iframe.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/utils'
 @ ./~/sockjs-client/lib/utils/iframe.js 1:0-114
 @ ./~/sockjs-client/lib/iframe-bootstrap.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/utils/browser-crypto.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/utils'
 @ ./~/sockjs-client/lib/utils/browser-crypto.js 1:0-114
 @ ./~/sockjs-client/lib/utils/random.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/sender/xdr.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/sender'
 @ ./~/sockjs-client/lib/transport/sender/xdr.js 1:0-107
 @ ./~/sockjs-client/lib/info-receiver.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/sender/xdr.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/sender'
 @ ./~/sockjs-client/lib/transport/sender/xdr.js 1:0-114
 @ ./~/sockjs-client/lib/info-receiver.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/lib/ajax-based.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/lib'
 @ ./~/sockjs-client/lib/transport/lib/ajax-based.js 1:0-107
 @ ./~/sockjs-client/lib/transport/xhr-streaming.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/lib/sender-receiver.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/lib'
 @ ./~/sockjs-client/lib/transport/lib/sender-receiver.js 1:0-107
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/sender/jsonp.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/sender'
 @ ./~/sockjs-client/lib/transport/sender/jsonp.js 1:0-107
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/sender/jsonp.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/sender'
 @ ./~/sockjs-client/lib/transport/sender/jsonp.js 1:0-114
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/iframe.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport'
 @ ./~/sockjs-client/lib/transport/iframe.js 1:0-107
 @ ./~/sockjs-client/lib/transport/lib/iframe-wrap.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/receiver/eventsource.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/receiver'
 @ ./~/sockjs-client/lib/transport/receiver/eventsource.js 1:0-107
 @ ./~/sockjs-client/lib/transport/eventsource.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/receiver/xhr.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/receiver'
 @ ./~/sockjs-client/lib/transport/receiver/xhr.js 1:0-107
 @ ./~/sockjs-client/lib/transport/xhr-streaming.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/receiver/jsonp.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/receiver'
 @ ./~/sockjs-client/lib/transport/receiver/jsonp.js 1:0-107
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/receiver/jsonp.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/receiver'
 @ ./~/sockjs-client/lib/transport/receiver/jsonp.js 1:0-114
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/receiver/htmlfile.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/receiver'
 @ ./~/sockjs-client/lib/transport/receiver/htmlfile.js 1:0-107
 @ ./~/sockjs-client/lib/transport/htmlfile.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/receiver/htmlfile.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/receiver'
 @ ./~/sockjs-client/lib/transport/receiver/htmlfile.js 1:0-114
 @ ./~/sockjs-client/lib/transport/htmlfile.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/browser/eventsource.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/browser'
 @ ./~/sockjs-client/lib/transport/browser/eventsource.js 1:0-114
 @ ./~/sockjs-client/lib/transport/eventsource.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/browser/websocket.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/browser'
 @ ./~/sockjs-client/lib/transport/browser/websocket.js 1:0-114
 @ ./~/sockjs-client/lib/transport/websocket.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/url-parse/lolcation.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/url-parse'
 @ ./~/url-parse/lolcation.js 1:0-114
 @ ./~/url-parse/index.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/lib/polling.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/lib'
 @ ./~/sockjs-client/lib/transport/lib/polling.js 1:0-107
 @ ./~/sockjs-client/lib/transport/lib/sender-receiver.js
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/lib/buffered-sender.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/lib'
 @ ./~/sockjs-client/lib/transport/lib/buffered-sender.js 1:0-107
 @ ./~/sockjs-client/lib/transport/lib/sender-receiver.js
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/browser/abstract-xhr.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/browser'
 @ ./~/sockjs-client/lib/transport/browser/abstract-xhr.js 1:0-114
 @ ./~/sockjs-client/lib/transport/sender/xhr-cors.js
 @ ./~/sockjs-client/lib/info-receiver.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/browser/abstract-xhr.js
Module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/sockjs-client/lib/transport/browser'
 @ ./~/sockjs-client/lib/transport/browser/abstract-xhr.js 1:0-107
 @ ./~/sockjs-client/lib/transport/sender/xhr-cors.js
 @ ./~/sockjs-client/lib/info-receiver.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ (webpack)-dev-server/client/socket.js
 @ (webpack)-dev-server/client?http://localhost:4200/
 @ multi main
Child html-webpack-plugin for "index.html":
    
    ERROR in Entry module not found: Error: Can't resolve '/Users/WhatAHell/Documents/!doesnotworkRnD/' in '/Users/WhatAHell/Documents/!doesnotworkRnD/!Property/no-peoperty/node_modules/angular-cli/models'
webpack: bundle is now VALID.
^Z
[1]+  Stopped                 ng serve`
> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

--- a/packages/angular-cli/commands/new.ts
+++ b/packages/angular-cli/commands/new.ts
@@ -1,5 +1,6 @@
 import * as chalk from 'chalk';
 import InitCommand from './init';
+import {oneLine} from 'common-tags';
 
 const Command = require('../ember-cli/lib/models/command');
 const Project = require('../ember-cli/lib/models/project');
@@ -36,6 +37,12 @@ const NewCommand = Command.extend({
       return Promise.reject(new SilentError(
         `The "ng ${this.name}" command requires a name argument to be specified. ` +
         `For more details, use "ng help".`));
+    }
+    if (!packageName.match(/^[a-zA-Z][.0-9a-zA-Z]*(-[a-zA-Z][.0-9a-zA-Z]*)*$/)) {
+      return Promise.reject(new SilentError(oneLine`
+        Project name "${packageName}" is not valid. New project names must
+        start with a letter, and must contain only alphanumeric characters or dashes.
+      `));
     }
 
     commandOptions.name = packageName;

--- a/tests/acceptance/new.spec.ts
+++ b/tests/acceptance/new.spec.ts
@@ -1,18 +1,17 @@
-'use strict';
+const fs = require('fs-extra');
+const ng = require('../helpers/ng');
+const existsSync = require('exists-sync');
+const expect = require('chai').expect;
+const forEach = require('lodash/forEach');
+const walkSync = require('walk-sync');
+const Blueprint = require('angular-cli/ember-cli/lib/models/blueprint');
+const path = require('path');
+const tmp = require('../helpers/tmp');
+const root = process.cwd();
+const util = require('util');
+const EOL = require('os').EOL;
+const SilentError = require('silent-error');
 
-var fs = require('fs-extra');
-var ng = require('../helpers/ng');
-var existsSync = require('exists-sync');
-var expect = require('chai').expect;
-var forEach = require('lodash/forEach');
-var walkSync = require('walk-sync');
-var Blueprint = require('angular-cli/ember-cli/lib/models/blueprint');
-var path = require('path');
-var tmp = require('../helpers/tmp');
-var root = process.cwd();
-var util = require('util');
-var EOL = require('os').EOL;
-var SilentError = require('silent-error');
 
 describe('Acceptance: ng new', function () {
   beforeEach(function () {
@@ -29,10 +28,10 @@ describe('Acceptance: ng new', function () {
 
   function confirmBlueprintedForDir(dir) {
     return function () {
-      var blueprintPath = path.join(root, dir, 'files');
-      var expected = walkSync(blueprintPath);
-      var actual = walkSync('.').sort();
-      var directory = path.basename(process.cwd());
+      let blueprintPath = path.join(root, dir, 'files');
+      let expected = walkSync(blueprintPath);
+      let actual = walkSync('.').sort();
+      let directory = path.basename(process.cwd());
 
       forEach(Blueprint.renamedFiles, function (destFile, srcFile) {
         expected[expected.indexOf(srcFile)] = destFile;
@@ -56,6 +55,33 @@ describe('Acceptance: ng new', function () {
     return confirmBlueprintedForDir('blueprints/ng2');
   }
 
+  it('requires a valid name (!)', () => {
+    return ng(['new', '!', '--skip-npm', '--skip-git', '--inline-template'])
+      .then(() => { throw new Error(); }, () => {});
+  });
+  it('requires a valid name (abc-.)', () => {
+    return ng(['new', 'abc-.', '--skip-npm', '--skip-git', '--inline-template'])
+      .then(() => { throw new Error(); }, () => {});
+  });
+  it('requires a valid name (abc-)', () => {
+    return ng(['new', 'abc-', '--skip-npm', '--skip-git', '--inline-template'])
+      .then(() => { throw new Error(); }, () => {});
+  });
+  it('requires a valid name (abc-def-)', () => {
+    return ng(['new', 'abc-def-', '--skip-npm', '--skip-git', '--inline-template'])
+      .then(() => { throw new Error(); }, () => {});
+  });
+  it('requires a valid name (abc-123)', () => {
+    return ng(['new', 'abc-123', '--skip-npm', '--skip-git', '--inline-template'])
+      .then(() => { throw new Error(); }, () => {});
+  });
+  it('requires a valid name (abc)', () => {
+    return ng(['new', 'abc', '--skip-npm', '--skip-git', '--inline-template']);
+  });
+  it('requires a valid name (abc-def)', () => {
+    return ng(['new', 'abc-def', '--skip-npm', '--skip-git', '--inline-template']);
+  });
+
   it('ng new foo, where foo does not yet exist, works', function () {
     return ng(['new', 'foo', '--skip-npm', '--skip-bower']).then(confirmBlueprinted);
   });
@@ -72,7 +98,7 @@ describe('Acceptance: ng new', function () {
     return ng(['new', 'FooApp', '--skip-npm', '--skip-bower', '--skip-git']).then(function () {
       expect(!existsSync('FooApp'));
 
-      var pkgJson = JSON.parse(fs.readFileSync('package.json', 'utf8'));
+      const pkgJson = JSON.parse(fs.readFileSync('package.json', 'utf8'));
       expect(pkgJson.name).to.equal('foo-app');
     });
   });
@@ -81,7 +107,7 @@ describe('Acceptance: ng new', function () {
     return ng(['new', 'FooApp', '--skip-npm', '--skip-bower', '--skip-git']).then(function () {
       expect(!existsSync('FooApp'));
 
-      var editorConfig = fs.readFileSync('.editorconfig', 'utf8');
+      const editorConfig = fs.readFileSync('.editorconfig', 'utf8');
       expect(editorConfig).to.exist;
     });
   });
@@ -93,7 +119,7 @@ describe('Acceptance: ng new', function () {
           throw new SilentError('Cannot run ng new, inside of ember-cli project should fail.');
         }, () => {
           expect(!existsSync('foo'));
-        })
+        });
       })
       .then(confirmBlueprinted);
   });
@@ -106,7 +132,7 @@ describe('Acceptance: ng new', function () {
 
   it('ng new with --dry-run does not create new directory', function () {
     return ng(['new', 'foo', '--dry-run']).then(function () {
-      var cwd = process.cwd();
+      const cwd = process.cwd();
       expect(cwd).to.not.match(/foo/, 'does not change cwd to foo in a dry run');
       expect(!existsSync(path.join(cwd, 'foo')), 'does not create new directory');
       expect(!existsSync(path.join(cwd, '.git')), 'does not create git in current directory');
@@ -116,14 +142,14 @@ describe('Acceptance: ng new', function () {
   it('ng new with --directory uses given directory name and has correct package name', function () {
     return ng(['new', 'foo', '--skip-npm', '--skip-bower', '--skip-git', '--directory=bar'])
       .then(function () {
-        var cwd = process.cwd();
+        const cwd = process.cwd();
         expect(cwd).to.not.match(/foo/, 'does not use app name for directory name');
         expect(!existsSync(path.join(cwd, 'foo')), 'does not create new directory with app name');
 
         expect(cwd).to.match(/bar/, 'uses given directory name');
         expect(existsSync(path.join(cwd, 'bar')), 'creates new directory with specified name');
 
-        var pkgJson = JSON.parse(fs.readFileSync('package.json', 'utf8'));
+        const pkgJson = JSON.parse(fs.readFileSync('package.json', 'utf8'));
         expect(pkgJson.name).to.equal('foo', 'uses app name for package name');
       });
   });

